Runtime/Interaction/InteractableObject.cs

```diff
@@ -87,11 +87,29 @@ public bool IsUsable
         protected override void Reset()
         {
             base.Reset();
-            
+
             // Sets the 'interactionLayerMask' to Default in order to not interact with Teleportation or UI rays.
             interactionLayerMask = 1;
         }
 
+        internal void OnTriggerEnter(Collider other)
+        {
+            SnapZone target = other.gameObject.GetComponent<SnapZone>();            
+            if (target != null && target.enabled && !IsInSocket)
+            {
+                target.AddHoveredInteractable(this);
+            }
+        }
+
+        internal void OnTriggerExit(Collider other)
+        {
+            SnapZone target = other.gameObject.GetComponent<SnapZone>();            
+            if (target != null && target.enabled)
+            {
+                target.RemoveHoveredInteractable(this);
+            }
+        }
```

Comment:
Not important but, can you make them virtual? We tend to complain a lot about Unity XR toolkit being so closed.

Reply:
As long as it stays internal, we can always make it virtual later.